signac/contrib/job.py

```diff
@@ -167,7 +167,11 @@ def ws(self):
         return self.workspace()
 
     def reset_statepoint(self, new_statepoint):
-        """Reset the state point of this job.
+        """Overwrite the state point of this job while preserving job data.
+
+        Change the job id if necessary.
+        For more, see
+        `Modifying the State Point <https://docs.signac.io/en/latest/jobs.html#modifying-the-state-point>`_.
```

Comment:
This should use an intersphinx reference to `_project-job-statepoint-modify`.

See this guide: https://github.com/glotzerlab/signac-docs/wiki/How-to-cross-reference-links-between-package-docs-and-signac-docs-using-intersphinx

Reply:
I'll make a new issue to add intersphinx because none of the files use it yet.

Reply:
made #452  to involve newer contributors